gapic/src/main/com/google/gapid/perfetto/views/GpuQueuePanel.java

```diff
@@ -108,6 +116,12 @@ public void renderTrack(RenderContext ctx, Repainter repainter, double w, double
 
         StyleConstants.Gradient color = gradient(data.titles[i].hashCode());
         color.applyBase(ctx);
+        // Link slices in Vulkan Event Panel and Gpu Queue Panel through submission id.
+        if (!selectedSIds.isEmpty() && i < sIds.length && !selectedSIds.contains(sIds[i])) {
+          ctx.setBackgroundColor(color.disabled); // Grey out unlinked ones.
+        } else if (i < sIds.length && selectedSIds.contains(sIds[i])) {
+          linkedIdxs.add(i);
+        }
```

Comment:
This could cause slices to be selected and grayed out. I think we should update this logic to be:
```
if (selected.contains(new Slice.Key(...)) || selectedSIds.contains(sIds[i])) {
  visibleSelected.add(new Highlight(...));
} else if (!selectedSIds.isEmpty()) {
  ctx.setBackgroundColor(color.disabled);
}
```

Note how that also gets rid of the `linkedIdxs` list. The `selected.contains(...)` part comes from below, but now would have to be done before drawing.

Reply:
Thanks for suggestion, it's cleaner after this change! Done in a new commit, Please take another look!